repair/hash.cc

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2018 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdexcept>
+#include "repair/hash.hh"
+#include "hashing.hh"
+#include "mutation_fragment.hh"
+#include "range_tombstone.hh"
+
+decorated_key_with_hash::decorated_key_with_hash(const schema& s, dht::decorated_key key, uint64_t seed) : dk(key) {
```

Comment:
This patch seems to make sense, but does it save compilation time? How much time does this small file take to compile?
If it doesn't take a long time, the split is not necessary. If it *does* take a long time, then probably there is a specific cause why - and we need to understand it, maybe we can improve this, and then we can improve the total build CPU usage, not just the highly-paralel compilation latency.